src/externalized/json/Json.cc

```diff
@@ -0,0 +1,207 @@
+#include "Json.h"
+
+JsonValue JsonValue::deserialize(const ST::string& str) {
+	auto r = RJsonValue_deserialize(str.c_str());
+	throwRustError(!r);
+	return JsonValue(r);
+}
+
+std::vector<JsonValue> JsonValue::toVec() const {
+	RustPointer<RJsonArray> array(RJsonValue_toArray(m_value.get()));
+	throwRustError(!array);
+	auto length = RJsonArray_length(array.get());
+	std::vector<JsonValue> vec;
+	vec.reserve(length);
+	for (size_t i = 0; i < length; i++) {
+		RustPointer<RJsonValue> val(RJsonArray_get(array.get(), i));
+		throwRustError(!val);
+		vec.push_back(val.release());
```

Comment:
Like in `JsonObject::keys()`, this should be `val.get()` or you'd leak the `RJsonValue` if `push_back` throws. This could also use `emplace_back`.

Reply:
Good catch!

Reply:
I just looked into this. And I think we should leave it as is, otherwise we would need to clone the RJsonValue again (which would be a second allocation just in this loop). My rationale behind this is that the `push_back` / `emplace_back` should only throw when reallocating in this case, but we already reserve the length of the vector beforehand, so reallocation should never be necessary.